src/api/include/migraphx/migraphx.hpp

```diff
@@ -317,6 +317,11 @@ struct interface_base : Base
             f();
             return migraphx_status_success;
         }
+        catch(const std::exception& ex)
+        {
+            std::cerr << "MIGraphX Error: " << ex.what() << std::endl;
```

Comment:
I think printing the error might be quite noisy, especially since this is used in `eliminate_contiguous` to decide if `contiguous` can be removed. It could also be confusing since there could be printouts of "MIGraphX Error" when there was no error.

We probably need to think of a better to transport these messages. Perhaps we store them in a thread-local variable? Or should we generate a `store_error_message` methods for interface handles?

Reply:
this is C++ API,  can you provide details on how it would catch exception from `eliminate_contiguous` ?

Reply:
> It could also be confusing since there could be printouts of "MIGraphX Error" when there was no error.

It would print whenever there is exception, Wouldn't exception mean there was something wrong or error ?

Reply:
> can you provide details on how it would catch exception from eliminate_contiguous ?

It removes `contiguous` as an input and then calls `compute_shape` if it throws an exception it will keep the `contiguous` operator. We dont want an "MIGraphX Error" on every try. 

> Wouldn't exception mean there was something wrong or error ?

Yes, but it could have been recovered so not a fatal error, and its not something a user would need to fix.

It could also lead QA assigning issues to us because it sees "MIGraphX Error" when the error is not from migraphx at all.